### PR TITLE
Simplify language identification using dictionary mapping

### PR DESCRIPTION
**Pull request by Arker123**
This PR suggests using a dictionary mapping approach for language identification.

### CODE REVIEW

**Review by williballenthin — COMMENTED** [review]
can't we just do `Language(args.language)`?

(need a try/except to handle missing cases)

**Comment by williballenthin** [comment]
while we're at it, we may want to derive the argparse choices from the Language enum.

File: floss/main.py [hunk 1]
```diff
@@ -198,8 +198,8 @@ def make_parser(argv):
     advanced_group.add_argument(
         "--language",
         type=str,
-        choices=[l.value for l in Language if l != Language.UNKNOWN] + ["none"],
-        default="",
+        choices=[lang.value for lang in Language],
+        default="none",
```
Comment (mr-tz) [inline floss/main.py:202]: does this include the unknown language in the usage output and allow to use "none"?

Comment (Arker123) [reply floss/main.py]: Yes, this change includes the unknown. I've updated it to allow "none" and not "Unknown" in latest commit 😄

File: floss/main.py [hunk 2]
```diff
@@ -548,15 +548,10 @@ def main(argv=None) -> int:
...
-    elif args.language == Language.RUST.value:
-        lang_id = Language.RUST
-    elif args.language == Language.DOTNET.value:
-        lang_id = Language.DOTNET
-    elif args.language == "none":
-        lang_id = Language.UNKNOWN
-    else:
+
+    try:
+        lang_id = Language(args.language)
```
Comment (mr-tz) [inline floss/main.py:553]: how can users disable language detection now?

**Comment by mr-tz** [comment]
now part of #901

**Pull request closed without merging**